StackView: Allow scaling of StackView window

its-izhar opened the pull request:
Implemented `width_scale` and `height_scale` options for stack view window.

This replaces the previous fixed scaling factors with user-configurable options, and provides default values if the user does not specify them. Values are expressed as a fraction of the total screen size.

Values are always clamped to be between > 0 and <= 1 to prevent invalid sizes. If out of range values are provided, defaults are used.

The implementation ensures that the stack view window is always centered on the screen.

The most important benefit of this is that the user can easily choose how big or small they want the window to be, and it will always be centered on the screen.

its-izhar committed:
StackView: Add size presets for stack view window

Replace width_scale/height_scale options with a simpler "size" preset
option. Users can now choose between "default" (85% width, 80% height)
and "large" (95% of screen).

Invalid size values fall back to "default" with a warning.

Signed-off-by: Izhar Ameer Shaikh <izharits@gmail.com>

its-izhar commented:
The main motivation is to be able to let the stackview window take more area of the screen. There is no easy way to do that because the window size is fixed in the code.

I personally use about 90-95% of the screen size in stackview to browse the new code repo when I don't have familiarity with the codebase to make most of the screen space. This commit does not change anything by default and only exposes the means so the window can be scaled by the user's preference. 

Let me know what you think. I had this commit in my fork for a bit, I thought others might benefit from it.

dhananjaylatkar commented on lua/cscope/stack_view/init.lua:
```diff
@@ -1,11 +1,22 @@
+-- width: fraction of screen width (0.85 = 85% of screen width)
+-- height: fraction of screen height (0.8 = 80% of screen height)
+-- Add more presets here as needed (e.g., "small", "medium")
+local size_presets = {
```
move it below opts

dhananjaylatkar reviewed (changes requested):
Thanks for revisiting the PR.

I have some suggestions.

its-izhar committed:
StackView: Address PR review feedback for size presets

Rename "default" preset to "medium", restore col gap, access preset
inside buf_open, and match previous default height of 0.7.

Signed-off-by: Izhar Ameer Shaikh <izharits@gmail.com>
